### PR TITLE
Prune remote reply tracking

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/accounts_test.go [hunk 1]
```diff
@@ -1406,6 +1406,139 @@ func TestAccountRequestReplyTrackLatency(t *testing.T) {
...
+	// Test with a responder on second server, srvB. but they will not respond.
+	cfoo, crFoo, _ := newClientForServer(srvB)
+	defer cfoo.nc.Close()
+	fooAcc, _ := srvB.LookupAccount("$foo")
+	if err := cfoo.registerWithAccount(fooAcc); err != nil {
+		t.Fatalf("Error registering client with 'foo' account: %v", err)
+	}
+
+	// Set new limits
+	fooAcc.SetAutoExpireTTL(time.Millisecond)
```
Comment (kozlovic) [inline server/accounts_test.go]: This seem really low for tests on Travis, no?

File: server/accounts_test.go [hunk 2]
```diff
@@ -1406,6 +1406,139 @@ func TestAccountRequestReplyTrackLatency(t *testing.T) {
...
+	fooAcc.SetMaxAutoExpireResponseMaps(5)
+
+	// Now setup the resonder under cfoo and the listener for the results
```
Comment (kozlovic) [inline server/accounts_test.go]: `responder`

File: server/accounts_test.go [hunk 3]
```diff
@@ -1406,6 +1406,139 @@ func TestAccountRequestReplyTrackLatency(t *testing.T) {
...
+	numTracking := tracking()
+
+	if numTracking != 2 {
+		t.Fatalf("Expected to have 2 tracking replies, got %d", numTracking)
+	}
+
+	// Make sure these remote tracking replies honor the current auto expire TTL.
+	time.Sleep(2 * time.Millisecond)
+
+	rc.mu.Lock()
```
Comment (kozlovic) [inline server/accounts_test.go]: This is really strange. You invoke the pruning manually here? I don't see a place where we remove those entries from rrTracking simply based on TTL (we do based on count).

File: server/accounts_test.go (hunk 1 above)
Comment (derekcollison) [reply server/accounts_test.go]: Not imer related, simply for the prune test.

File: server/accounts_test.go (hunk 2 above)
Comment (derekcollison) [reply server/accounts_test.go]: Will fix..

File: server/accounts_test.go (hunk 3 above)
Comment (derekcollison) [reply server/accounts_test.go]: Correct, same as other auto-expire, we remove when there is pressure and we hit a threshold. I need to rework but wanted to make this consistent with the others for now.

**Comment by derekcollison** [comment]
Will look at race.

**Pull request merged**